build/build.go

```diff
@@ -574,6 +575,20 @@ func NewMappingCallback(m *sourcemap.Map, goroot, gopath string) func(generatedL
 	}
 }
 
+func jsFilesFromDir(dir string) ([]string, error) {
+	files, err := ioutil.ReadDir(dir)
```

Comment:
I just wanted to comment here. I realize it is not a regression since the code contained `ioutil.ReadDir(pkg.Dir)` previously.

But this looks like it won't work in environments where direct disk access isn't available (e.g., the Go playground, or when using GopherJS as a library to compile in-memory without writing to disk).

It'd be outside the scope of this PR, but are there plans to try to make this operate/support virtual filesystems and avoid having duplicate code/logic for building with gopherjs-the-binary vs. gopherjs-the-library?